package/MDAnalysis/analysis/base.py

```diff
@@ -40,17 +40,59 @@
 logger = logging.getLogger(__name__)
 
 
+class Results(dict):
+    r"""Container object for storing results.
+
+    Results are extend dictionaries by enabling values to be accessed by key,
+    `results["value_key"]`, or by an attribute, `results.value_key`. 
+    They store all results obatined from an analysis after the ``run`` call.
+
+    The current implementation is similar to the `Bunch` class in `sklearn`.
+
+    Examples
+    --------
+    >>> results = Results(a=1, b=2)
+    >>> results['b']
+    2
+    >>> results.b
+    2
+    >>> results.a = 3
+    >>> results['a']
+    3
+    >>> results.c = [1, 2, 3, 4]
+    >>> results['c']
+    [1, 2, 3, 4]
+    """
+
+    def __init__(self, **kwargs):
+        super().__init__(kwargs)
+
+    def __setattr__(self, key, value):
+        self[key] = value
+
+    def __dir__(self):
+        return self.keys()
+
+    def __getattr__(self, key):
+        if key in self.keys():
```

Comment:
This should probably be a try/except, otherwise you're looking up the key twice.

Reply:
Good point. I changed this.